src/oci/runc.rs

```diff
@@ -0,0 +1,299 @@
+/// DefaultRuncExecCommand is a wrapper which can be used to execute runc in a standard way.
+struct DefaultRuncExecCommand;
+
+impl ExecCommand for DefaultRuncExecCommand {}
+
+#[async_trait]
+trait ExecCommand: Debug + DynClone + Send + Sync {
+    /// Run a command and return its `Output`.
+    async fn run_output(&self, binary: &Path, cmd: &OsStr, args: Vec<String>) -> Result<Output> {
+        Command::new(binary)
+            .arg(cmd)
+            .args(args)
+            .output()
+            .await
+            .context("run conmon")
+    }
+}
+
+clone_trait_object!(ExecCommand);
+
+#[derive(AsRefStr, Clone, Debug, Hash, Eq, PartialEq)]
+#[strum(serialize_all = "kebab_case")]
+/// Available global arguments for runc.
+pub enum RuncGlobalArgs {
+    /// Debug
+    Debug,
+
+    /// Log file path
+    Log(PathBuf),
+
+    /// Log Format {test | json}
+    LogFormat(String),
+
+    /// 
+    Root(PathBuf),
+    Criu(String),
+    SystemdCgroup(String),
+    Rootless(String),
+    Version,
+}
+
+impl fmt::Display for RuncGlobalArgs {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use crate::oci::runc::RuncGlobalArgs::*;
+        write!(f, "--")?;
+
+        fn write_kv<K, V>(f: &mut fmt::Formatter<'_>, key: K, value: V) -> fmt::Result
+        where
+            K: AsRef<str>,
+            V: fmt::Display,
+        {
+            write!(f, "{}={}", key.as_ref(), value)
+        }
+
+        match self {
+            Log(path) => write_kv(f, self, path.display()),
+            Root(path) => write_kv(f, self, path.display()),
+            LogFormat(format) => write_kv(f, self, format),
+            Criu(criu) => write_kv(f, self, criu),
+            SystemdCgroup(cgroup) => write_kv(f, self, cgroup),
+            Rootless(rootless) => write_kv(f, self, rootless),
+            _ => write!(f, "{}", self.as_ref()),
+        }
+    }
+}
+
+
+#[derive(AsRefStr, Clone, Debug, Hash, Eq, PartialEq)]
+#[strum(serialize_all = "kebab_case")]
+/// Available arguments for 'runc checkpoint'.
+pub enum RuncCheckpointArgs {
+    /// Runc Grlobal Args
+    RuncGlobalArgs(RuncGlobalArgs),
+
+    ImagePath(PathBuf),
+    WorkPath(PathBuf),
+    ParentPath(PathBuf),
+    LeaveRunning,
+    TcpEstablished,
+    ExtUnixSk,
+    ShellJob,
+    LazyPages,
+    StatusFd(String),
+    PageServer(String),
+    FileLocks,
+    PreDump,
+    ManageCgroupsMode(String),
+    EmptyNs(String),
+    AutoDedup,
+}
+
+impl fmt::Display for RuncCheckpointArgs {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use crate::oci::runc::RuncCheckpointArgs::*;
+        write!(f, "--")?;
+
+        fn write_kv<K, V>(f: &mut fmt::Formatter<'_>, key: K, value: V) -> fmt::Result
+        where
+            K: AsRef<str>,
+            V: fmt::Display,
+        {
+            write!(f, "{}={}", key.as_ref(), value)
+        }
+
+        match self {
+            RuncGlobalArgs(args) => write_kv(f, self, args),
```

Comment:
I'm not even sure this is valid, basically I just want to process the `RuncGlobalArgs` how ever they should be processed, as the global args are global and can be added too any sub command. anyhow thoughts?

Reply:
I think it does not matter where to add the global args, right? So this seems fine to me…